esp-wifi/src/wifi/mod.rs

```diff
@@ -262,41 +283,65 @@ pub struct AccessPointConfiguration {
     pub secondary_channel: Option<u8>,
 
     /// The set of protocols supported by the access point.
-    #[cfg_attr(feature = "defmt", defmt(Debug2Format))]
     pub protocols: EnumSet<Protocol>,
 
     /// The authentication method to be used by the access point.
     pub auth_method: AuthMethod,
 
     /// The password for securing the access point (if applicable).
-    pub password: heapless::String<64>,
+    pub password: alloc::string::String,
 
     /// The maximum number of connections allowed on the access point.
     pub max_connections: u16,
 }
 
+#[cfg(feature = "defmt")]
+impl defmt::Format for AccessPointConfiguration {
```

Comment:
Do you have something against `derive(defmt::Format)`? :D you can do things like this:

```rust
#[derive(Format)]
struct Reading {
    #[defmt(Display2Format)]
    temperature: Celsius,
}
```

according to the [book](https://defmt.ferrous-systems.com/format)

Reply:
Those annotations pull in `core::fmt` machinery, according to https://defmt.ferrous-systems.com/format#uncompressed-adapters. I'd prefer the manual impl here I think.

Reply:
Okay the only one left has a TODO which can be resolved with a new defmt version